sql/sql_connect.cc

```diff
@@ -255,6 +262,105 @@ void time_out_user_resource_limits(THD *thd, USER_CONN *uc)
   DBUG_VOID_RETURN;
 }
 
+static uint64 get_connection_delay_time(uint32 count)
+{
+  /*
+   * Based on that invalid password errors will cause user can't connect server
+  ever and server close connection if it got message from client after
+  wait_timeout
+   * so,
+   step = read_timeout/(max_password_errors - password_errors_before_delay)
+  delay = step *(failed_count  -password_errors_before_delay)
+*/
+  /* Time unit for delay step is seconds, convert to milliseconds*/
+  uint64 read_timeout= global_system_variables.net_read_timeout * 1000;
+  uint64 step=
+      read_timeout / (max_password_errors - password_errors_before_delay);
+  /* If max_password_errors is large, step is very small which
+   * influence nothing, so 1 second is minim delay step */
+  step= step > 1000 ? step : 1000;
```

Comment:
thinking more about it, I believe this is not a good idea. Yes, the delay will be growing slowly if max_password_errors is big. But with 1s step, the delay will very quickly reach connect_timeout (for example, if connect_timeout is 10, then the delay will be 10s after 10 wrong passwords) and then it will not grow **at all**. Constant linear growth seems to be more adequate here